src/core/CMakeLists.txt

```diff
@@ -1000,6 +1009,8 @@ INCLUDE_DIRECTORIES(
   metadata
   pal
   processing
+  providers
```

Comment:
There's a couple of reasons

My main reason was I needed the memory provider utils part. And then it seemed odd that we'd have utilities for working with certain providers in core when those providers themselves aren't in core.

Then it became an exercise in seeing what was missing from enabling plugin based providers...